packages/uma-auth-client/src/steps/ConnectUma.tsx

```diff
@@ -12,14 +12,15 @@ export const ConnectUma = ({ setStep }: { setStep: (step: Step) => void }) => {
   const codeChallenge = "1234";
   const codeChallengeMethod = "S256";
 
-  const [uma, setUma] = useState("");
+  const { uma, setUma } = useUser();
 
   const handleChangeUma = (value: string) => {
     setUma(value);
   };
 
   const handleConnectYourUMA = () => {
-    window.location.href = `http://localhost:3000/apps/new?client_id=${clientId}&redirect_uri=${redirectUri}&response_type=${responseType}&code_challenge=${codeChallenge}&code_challenge_method=${codeChallengeMethod}&required_commands=send_payments&optional_commands=read_balance,read_transactions&budget=10.USD%2Fmonthly&expiration_period=year`;
+    // window.location.href = `http://localhost:3000/apps/new?client_id=${clientId}&redirect_uri=${redirectUri}&response_type=${responseType}&code_challenge=${codeChallenge}&code_challenge_method=${codeChallengeMethod}&required_commands=send_payments&optional_commands=read_balance,read_transactions&budget=10.USD%2Fmonthly&expiration_period=year`;
```

Comment:
What's the plan here? Should we remove this? I was actually kind of confused on why we need a loading indicator if we're going to be redirecting to the sign in flow and back. Can you explain the intended end-state?

Reply:
I think the idea was we open in a new tab the permissions page/vasp login and have the page close. Not sure if there's an easy way of getting the updated info but it would also not be great to lose context on the original page imo (like if you're at a checkout page, or watching/listening/consuming some content). If we store the connection details in localStorage it's possible we can just poll for that upon redirect/closing the other tab?

Reply:
Yeah this is the tricky part of oauth... it generally operates through redirects in order to get the auth code back to the calling client app. Opening up a new window wouldn't allow that exactly.

Reply:
for posterity, talked offline and we decided to start with just plain redirects. There's a proposal for OAuth Web Message which handles a case like this but would need to go through security review and this can be a follow up improvement